rpc/handlers.go

```diff
@@ -381,11 +398,24 @@ func (h *Handler) TransactionReceiptByHash(hash felt.Felt) (*TransactionReceipt,
 	}
 
 	var receiptBlockNumber *uint64
+	status := StatusAcceptedL2
+
 	if blockHash != nil {
 		receiptBlockNumber = &blockNumber
+		l1Head, err := h.bcReader.L1Head()
+		if err != nil && !errors.Is(err, db.ErrKeyNotFound) {
+			return nil, ErrInternal
+		}
+
+		if l1Head != nil && l1Head.BlockNumber >= *receiptBlockNumber {
+			status = StatusAcceptedL1
+		}
```

Comment:
Let's make this a helper?

`bcReader.BlockIsL1Verified(blockHash)`

Reply:
Sounds good

Reply:
Done